### PR TITLE
[GlobalISel] Import llvm.stepvector

### PR DESCRIPTION
**Pull request by tschuett**

### CODE REVIEW

File: llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp [hunk 1]
```diff
@@ -2597,6 +2597,10 @@ bool IRTranslator::translateKnownIntrinsic(const CallInst &CI, Intrinsic::ID ID,
     return translateExtractVector(CI, MIRBuilder);
   case Intrinsic::vector_insert:
     return translateInsertVector(CI, MIRBuilder);
+  case Intrinsic::stepvector: {
+    MIRBuilder.buildStepVector(getOrCreateVReg(CI), 1);
```
Comment (arsenm) [inline llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp:2601]: Not sure why this has an operand the intrinsic doesn't but it's already that way. I'm confused by the DAG API, it's one level removed but doing the same thing

**Review by arsenm — APPROVED** [review]

File: llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp (hunk 1 above)
Comment (tschuett) [reply llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp:2601]: The LLVM-IR intrinsics has an implicit factor of one:
https://llvm.org/docs/LangRef.html#llvm-stepvector-intrinsic
For GlobalISel, we have a real factor:
https://github.com/llvm/llvm-project/pull/115598

G_STEP_VECTOR is a cheap name for the AArch64 
INDEX (immediates)
instruction.

We and the DAG did the same for the vscale intrinsic:
https://llvm.org/docs/GlobalISel/GenericOpcode.html#g-vscale
The standard vscale pattern takes a scale, while the LLVM-IR intrinsic doesn't:
`def : Pat<(vscale GPR64:$scale), (MADDXrrr (UBFMXri (RDVLI_XI 1), 4, 63), $scale, XZR)>;`

Comment (tschuett) [reply llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp:2601]: The difference is probably between LLVM-IR and hardware instructions.

Comment (tschuett) [reply llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp:2601]: Slightly above:
```
  case Intrinsic::vscale: {
    MIRBuilder.buildVScale(getOrCreateVReg(CI), 1);
    return true;
  }
```

**Pull request merged**